LGTM. @Gowa2017, could you take care of the warnings reported by the Continuous Integration tests, please?
```
 ndpiSimpleIntegration.c: In function ‘ndpi_process_packet’:
ndpiSimpleIntegration.c:729:20: error: comparison of integer expressions of different signedness: ‘uint32_t’ {aka ‘unsigned int’} and ‘int’ [-Werror=sign-compare]
  729 |   if (thread_index != reader_thread->array_index) {
      |                    ^~
cc1: all warnings being treated as errors
```

--- a/example/ndpiSimpleIntegration.c
+++ b/example/ndpiSimpleIntegration.c
@@ -137,7 +137,7 @@ struct nDPI_workflow {
 struct nDPI_reader_thread {
   struct nDPI_workflow * workflow;
   pthread_t thread_id;
-  int array_index;
+  uint32_t array_index;
 };
 
 static struct nDPI_reader_thread reader_threads[MAX_READER_THREADS] = {};

